### PR TITLE
remove GAX test code from BOM

### PR DESCRIPTION
**Pull request by elharo**
@suztomo

### CODE REVIEW

File: boms/cloud-oss-bom/pom.xml [hunk 1]
```diff
@@ -211,11 +211,25 @@
        </dependency>
        <dependency>
          <groupId>com.google.api</groupId>
-         <artifactId>gax-bom</artifactId>
+         <artifactId>gax</artifactId>
          <version>${gax.version}</version>
          <type>pom</type>
          <scope>import</scope>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:217]: The artifact gax is not a pom.

**Comment by suztomo** [comment]
gax's testlib-classifier artifact contains classes useful to write tests for gax (test infrastructure). No unit tests any more (I read your https://github.com/googleapis/gax-java/issues/700).

It's not for GCP customers, but for Google library maintainers to write unit tests with GAX. That's why this PR is removing the testlib-classifier artifact. Is this correct?

```
suztomo-macbookpro44% jar tf gax-1.59.0-testlib.jar 
META-INF/
META-INF/MANIFEST.MF
com/
com/google/
com/google/api/
com/google/api/gax/
com/google/api/gax/rpc/
com/google/api/gax/rpc/testing/
com/google/api/gax/rpc/testing/FakeStatusCode.class
com/google/api/gax/rpc/testing/FakePagedApi$ListIntegersPagedResponse.class
com/google/api/gax/rpc/testing/FakeSimpleApi$StashCallable.class
com/google/api/gax/rpc/testing/FakePagedApi$ListIntegersPagedResponseFactory.class
com/google/api/gax/rpc/testing/FakeSimpleApi.class
com/google/api/gax/rpc/testing/FakeClientSettings.class
com/google/api/gax/rpc/testing/MockStreamingApi.class
com/google/api/gax/rpc/testing/FakeBatchableApi$LabeledIntList.class
com/google/api/gax/rpc/testing/FakeStreamingApi$ClientStreamingStashCallable$AccumulatingStreamObserver.class
com/google/api/gax/rpc/testing/FakeTransportChannel.class
com/google/api/gax/rpc/testing/FakeClientSettings$1.class
com/google/api/gax/rpc/testing/FakeStreamingApi$BidiStreamingStashCallable.class
com/google/api/gax/rpc/testing/FakeStreamingApi$BidiStreamingStashCallable$AccumulatingClientStream.class
com/google/api/gax/rpc/testing/FakePagedApi$ListIntegersSizedPage.class
com/google/api/gax/rpc/testing/FakeOperationSnapshot.class
com/google/api/gax/rpc/testing/MockStreamingApi$MockServerStreamingCallable.class
com/google/api/gax/rpc/testing/FakeStubSettings$1.class
com/google/api/gax/rpc/testing/FakeBatchableApi$SquarerBatchingDescriptor$1.class
com/google/api/gax/rpc/testing/FakeBatchableApi$SquarerBatchingDescriptorV2.class
com/google/api/gax/rpc/testing/FakeChannel.class
com/google/api/gax/rpc/testing/FakeBatchableApi$LabeledIntSquarerCallable.class
com/google/api/gax/rpc/testing/FakeStreamingApi$ClientStreamingStashCallable.class
com/google/api/gax/rpc/testing/FakeOperationApi.class
com/google/api/gax/rpc/testing/FakeApiException.class
com/google/api/gax/rpc/testing/FakeStreamingApi$ServerStreamingStashCallable.class
com/google/api/gax/rpc/testing/MockStreamingApi$MockResponseObserver.class
com/google/api/gax/rpc/testing/FakeOperationApi$FakeOperation.class
com/google/api/gax/rpc/testing/FakeOperationSnapshot$Builder.class
com/google/api/gax/rpc/testing/FakeStreamingApi.class
com/google/api/gax/rpc/testing/FakePagedApi.class
com/google/api/gax/rpc/testing/AutoValue_FakeOperationSnapshot.class
com/google/api/gax/rpc/testing/MockStreamingApi$MockServerStreamingCall.class
com/google/api/gax/rpc/testing/FakeStatusException.class
com/google/api/gax/rpc/testing/FakePagedApi$PagedStashCallable.class
com/google/api/gax/rpc/testing/FakeBatchableApi$SquarerBatchingDescriptorV2$1.class
com/google/api/gax/rpc/testing/FakePagedApi$StashPagedListDescriptor.class
com/google/api/gax/rpc/testing/FakeBatchableApi.class
com/google/api/gax/rpc/testing/FakePagedApi$ListIntegersPagedResponse$1.class
com/google/api/gax/rpc/testing/MockStreamingApi$MockStreamController.class
com/google/api/gax/rpc/testing/FakeBatchableApi$SquarerBatchingDescriptor.class
com/google/api/gax/rpc/testing/AutoValue_FakeOperationSnapshot$Builder.class
com/google/api/gax/rpc/testing/FakeExceptionCallable.class
com/google/api/gax/rpc/testing/FakePagedApi$1.class
com/google/api/gax/rpc/testing/FakePagedApi$ListIntegersPage.class
com/google/api/gax/rpc/testing/FakeStubSettings.class
com/google/api/gax/rpc/testing/FakeStubSettings$Builder.class
com/google/api/gax/rpc/testing/FakeOperationApi$OperationStashCallable.class
com/google/api/gax/rpc/testing/FakeExceptionCallable$ExceptionTransformingFuture.class
com/google/api/gax/rpc/testing/FakeClientSettings$Builder.class
com/google/api/gax/rpc/testing/FakeCallableFactory.class
com/google/api/gax/rpc/testing/FakeOperationApi$FakeOperationFuture.class
com/google/api/gax/rpc/testing/FakeCallContext.class
com/google/api/gax/rpc/testing/AutoValue_FakeOperationSnapshot$1.class
com/google/api/gax/rpc/testing/FakeStreamingApi$1.class
com/google/api/gax/rpc/testing/FakeStreamingApi$StreamControllerStash.class
```

File: boms/cloud-oss-bom/pom.xml (hunk 1 above)
Comment (elharo) [reply boms/cloud-oss-bom/pom.xml]: good catch; fixed

**Review by elharo — COMMENTED** [review]
testlib shouldn't be used outside the project itself, any more than anything else in src/test. It's not documented, not subject to versioning constraints or backwards compatibility, and arguably shouldn't be published on maven central.

**Review by suztomo — APPROVED** [review]

**Pull request merged**